packages/@aws-cdk/aws-glue/lib/security-configuration.ts

```diff
@@ -0,0 +1,239 @@
+import * as kms from '@aws-cdk/aws-kms';
+import * as cdk from '@aws-cdk/core';
+import * as constructs from 'constructs';
+import { CfnSecurityConfiguration } from './glue.generated';
+
+/**
+ * Interface representing a created or an imported {@link SecurityConfiguration}.
+ */
+export interface ISecurityConfiguration extends cdk.IResource {
+  /**
+   * The name of the security configuration.
+   * @attribute
+   */
+  readonly securityConfigurationName: string;
+}
+
+/**
+ * Attributes for importing {@link SecurityConfiguration}.
+ */
+export interface SecurityConfigurationAttributes {
+  /**
+   * The name of the security configuration.
+   */
+  readonly securityConfigurationName: string;
+}
+
+/**
+ * Encryption mode for S3.
+ */
+export enum S3EncryptionMode {
+  /**
+   * No encryption.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * Server side encryption (SSE) with an Amazon S3-managed key.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html
+   */
+  S3_MANAGED = 'SSE-S3',
+
+  /**
+   * Server-side encryption (SSE) with an AWS KMS key managed by the account owner.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+   */
+  KMS = 'SSE-KMS',
+}
+
+/**
+ * Encryption mode for CloudWatch Logs.
+ */
+export enum CloudWatchEncryptionMode {
+  /**
+   * No encryption.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * Server-side encryption (SSE) with an AWS KMS key managed by the account owner.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+   */
+  KMS = 'SSE-KMS',
+}
+
+/**
+ * Encryption mode for Job Bookmarks.
+ */
+export enum JobBookmarksEncryptionMode {
+  /**
+   * No encryption.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * Client-side encryption (CSE) with an AWS KMS key managed by the account owner.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingClientSideEncryption.html
+   */
+  CLIENT_SIDE_KMS = 'CSE-KMS',
+}
+
+/**
+ * S3 encryption configuration.
+ */
+export interface S3Encryption {
+  /**
+   * Encryption mode.
+   */
+  readonly mode: S3EncryptionMode,
+
+  /**
+   * The KMS key to be used to encrypt the data.
+   * @default no kms key if mode = DISABLED or S3_MANAGED. A key must be provided for KMS mode.
+   */
+  readonly kmsKey?: kms.IKey,
```

Comment:
There are a few things to consider here
1. We have 3 encryption configs that all can potentially require a kms key
2. Do we create 1 key per config or 1 for use when any is absent?
3. From a UX standpoint, if we expose the kms key as a property - whether 3 or 1 - what abt if the user provides a key for some configs and no key for others?

To have a friendly construct, I like the one key option... it doesn't lead to any surprising behaviours, but it limits the construct options compared to CF one... if one key, we can even simplify props to accept an optional key - rather than 3 for each config type and create one if not needed.

What do you think?

Reply:
I don't want to limit the constructs capabilities - so the inputs are ok as is. I think it's also reasonable to create a single shared key for the configs that are missing a required key.
 
```ts
const securityConfig = new SecurityConfiguration(...);

// each of these either return the key that was passed, or the shared key that was created.
// can also return undefined if the relevant encryption wasn't configured 
securityConfig.cloudWatchEncryptionKey;
securityConfig.s3EncryptionKey;
securityConfig.jobBookmarksEncryptionKey;
```

BTW - are you sure the key is required in CFN? The docs mention otherwise but I recognize it might just be that way in the docs because the entire config is optional.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-securityconfiguration-cloudwatchencryption.html#cfn-glue-securityconfiguration-cloudwatchencryption-kmskeyarn

Reply:
I've updated the PR with one take on this (single default key that gets created and used if any config requires a kms key if none is provided)

Reply:
Just saw ur comment about exposing 3 properties instead of 1. I like ur suggestion, it allows the construct user to completely rely on the construct for retrieving all keys rather than doing external house keeping.

I'll have another go at this.